### PR TITLE
added details, deathdate, hair color, weight to performers and added details to studios

### LINKED ISSUE #545: [Feature] Date of death field for performers
**Is your feature request related to a problem? Please describe.**
Incorrect age is reported for dead performers

**Describe the solution you'd like**
* Add a field for date of death to correctly calculate performer's current age.
* Add death scraping capability to freeone's scraper.
  * Freeones lists the date of death, "Passed away on December 5, 2017", under the birth date. Example:  https://www.freeones.com/august-ames/profile

### LINKED ISSUE #712: [Feature] Biography field for performers
**Is your feature request related to a problem? Please describe.**
When scraping some sites I found well written bios / descriptions for performers. It would be nice to hoard this data.

**Describe the solution you'd like**
Add a bio field to performers.

### LINKED ISSUE #811: [Feature] Description fields for Performers and Studios
**Is your feature request related to a problem? Please describe.**
When updating Studio information I want to make notes which will be attached to the studio page. Scenes have this already implimented as a description field.

**Describe the solution you'd like**
When editing a Studio or Performer a description box will be available to write text data to.

### PR DESCRIPTION
**Pull request by julien0221**
I added a new column details on the table performers and studios.
I also changed the scrapper freeones to get the bio from the freeones's website.
I added it on is missing filter.
Resolves #811
Resolves #712 
Resolves #545

### CODE REVIEW

**Comment by WithoutPants** [comment]
Since it's doing a schema update for performers, perhaps consider including the following:
* Date of death (#545)
* Weight (https://github.com/stashapp/stash/issues/124#issuecomment-625946586)

**Comment by jeremymeyers** [comment]
If we include Weight we should include Hair Color.

**Comment by WithoutPants** [comment]
> If we include Weight we should include Hair Color.

Agree. Didn't see that in the github issues, but it is commonly asked for.

**Comment by julien0221** [comment]
Are you as well included rating into performers's table as well?

**Comment by WithoutPants** [comment]
No. Rating needs more changes outside the basics, so should be raised as a separate PR.

**Comment by julien0221** [comment]
Ok it is all done, I added deathdate, hair color and weight to performers. But one question before pushing, do you want me to add a filter death year into performers filters?

**Comment by WithoutPants** [comment]
Yes, filter criteria should be added for death date, hair colour and weight.

**Comment by julien0221** [comment]
ok so I added deathdate, weight and hair color to performers, and even on filters. 
I changed freeone's scraper to download the new data.

**Comment by bnkai** [comment]
There is currently no validation for the age. Could we add a simple one to make sure death_date > birthdate for the age display?

**Comment by julien0221** [comment]
I added a validation for the date. And I added some tests.

File: graphql/documents/data/performer.graphql [hunk 1]
```diff
@@ -29,4 +29,8 @@ fragment PerformerData on Performer {
     stash_id
     endpoint
   }
+  details
+  deathdate
```
Comment (WithoutPants) [inline graphql/documents/data/performer.graphql:33]: Should be `death_date`. Please fix in other graphql files as well.

I realise that this looks inconsistent with `birthdate`, but birthdate is a single (compound) word, death date is not.

File: graphql/schema/types/performer.graphql [hunk 2]
```diff
@@ -33,6 +33,10 @@ type Performer {
   scene_count: Int # Resolver
   scenes: [Scene!]!
   stash_ids: [StashID!]!
+  details: String
+  deathdate: String
+  hair_color: String
+  weight: String
```
Comment (WithoutPants) [inline graphql/schema/types/performer.graphql:39]: I think we should consider making this numeric rather than string, so that we can sort by it properly.

We probably also need to consider what unit to use. It may be worth separating this out to another PR if you want to kick this can down the road.

File: pkg/api/resolver_mutation_performer.go [hunk 3]
```diff
@@ -83,6 +84,24 @@ func (r *mutationResolver) PerformerCreate(ctx context.Context, input models.Per
...
 	}
+	if input.Details != nil {
+		newPerformer.Details = sql.NullString{String: *input.Details, Valid: true}
+	}
+	if input.Deathdate != nil {
+		newPerformer.Deathdate = models.SQLiteDate{String: *input.Deathdate, Valid: true}
+
+		if *input.Deathdate != "" {
+			if utils.IfDateHigherThanAnotherDate(*input.Birthdate, *input.Deathdate) {
+				return nil, errors.New("The date of Death should be higher than the date of Birth")
```
Comment (WithoutPants) [inline pkg/api/resolver_mutation_performer.go:95]: Declare an error at the top of the file and use that instead. 

Error message should be all lowercase.

File: pkg/api/resolver_mutation_performer.go [hunk 4]
```diff
@@ -153,6 +172,12 @@ func (r *mutationResolver) PerformerUpdate(ctx context.Context, input models.Per
 		updatedPerformer.Checksum = &checksum
 	}
 
+	if *input.Deathdate != "" {
```
Comment (WithoutPants) [inline pkg/api/resolver_mutation_performer.go:175]: This is going to panic if `Deathdate` is nil.

File: pkg/api/resolver_mutation_performer.go [hunk 5]
```diff
@@ -153,6 +172,12 @@ func (r *mutationResolver) PerformerUpdate(ctx context.Context, input models.Per
...
+		if utils.IfDateHigherThanAnotherDate(*input.Birthdate, *input.Deathdate) {
```
Comment (WithoutPants) [inline pkg/api/resolver_mutation_performer.go:176]: A `utils` function is not necessary. Create a local function (something like `validateDeathDate`) that converts the strings into dates and basically does what the `utils` function is doing.

**Review by WithoutPants — CHANGES_REQUESTED** [review]
The weight field requires more design decisions. I'm not sure it's worth including in this PR.

**Review by WithoutPants — APPROVED** [review]
